apps/admin/frontend/src/components/write_in_adjudication_button.tsx

```diff
@@ -0,0 +1,133 @@
+}: {
+  caption?: React.ReactNode;
+  isFocused: boolean;
+  isSelected: boolean;
+  hasInvalidEntry: boolean;
+  // newVal can be name of new or existing candidate,
+  // keyword 'invalid' for invalid write-in, or undefined
+  onChange: (newVal?: string) => void;
+  onInputBlur: () => void;
+  onInputFocus: () => void;
+  toggleVote: () => void;
+  value: string;
+  officialCandidateNames: string[];
+  writeInCandidateNames: string[];
+}): JSX.Element {
+  const theme = useTheme();
+  const [inputValue, setInputValue] = useState('');
+  const inputValueLowerCase = inputValue.toLowerCase();
+
+  function onKeyPress(val?: string) {
+    return setInputValue(val || '');
+  }
+
+  const filteredCandidateOptions = inputValue
+    ? officialCandidateNames.filter((name) =>
+        name.toLowerCase().includes(inputValueLowerCase)
+      )
+    : officialCandidateNames;
+  const filteredWriteInCandidateOptions = inputValue
+    ? writeInCandidateNames.filter((name) =>
+        name.toLowerCase().includes(inputValueLowerCase)
+      )
+    : writeInCandidateNames;
+  const options = filteredWriteInCandidateOptions
+    .concat(filteredCandidateOptions)
+    .map((name) => ({
+      label: name,
+      value: name,
+    }));
+
+  // 'Add: NEW_CANDIDATE' entry if there is no exact match
+  if (
+    inputValue &&
+    !options.find(
+      (item) => item.label.toLowerCase() === inputValue.toLowerCase()
+    )
+  ) {
+    options.push({ label: `Add: ${inputValue}`, value: inputValue });
+  }
+
+  // If value has been entered and it is a new entry, add it the dropdown
+  if (value && !options.find((option) => option.label === value)) {
+    options.push({ label: value, value });
+  }
+
+  if (!inputValue) {
+    options.unshift({ label: 'Not a mark', value: 'invalid' });
+  }
+
+  return (
+    <Container style={{ zIndex: isFocused ? 10 : 0 }}>
+      <CheckboxButton
+        isChecked={isSelected}
+        label="Write-in"
+        onChange={toggleVote}
+        style={{ borderRadius: '0.5rem 0.5rem 0 0' }}
+      />
+      <SearchSelect
+        aria-label="Select or add write-in candidate"
+        // The inner input does not clear the previous value when a
+        // double vote entry is detected because the `value` prop never
+        // changes. `hasInvalidEntry` as the key forces a re-render
+        key={`${hasInvalidEntry}`}
+        menuPortalTarget={document.body}
+        options={options}
+        onBlur={onInputBlur}
+        onFocus={onInputFocus}
+        onInputChange={onKeyPress}
+        onChange={(val) => {
+          setInputValue('');
+          onChange(val);
+        }}
+        value={value}
+        placeholder={
+          isFocused ? (
+            'Search or add...'
+          ) : (
+            <React.Fragment>
+              <Icons.Warning
+                color="warning"
+                style={{ marginRight: '0.5rem' }}
+              />
+              {isSelected ? 'Adjudicate' : 'Unmarked'} Write-in
```

Comment:
Consider using the full strings, which makes it easier to translate (if we ever do) and to search the codebase for (even if we don't translate).
```suggestion
              {isSelected ? 'Adjudicate Write-in' : 'Unmarked Write-in'}
```